addons/stock/models/stock_picking.py

```diff
@@ -941,6 +943,36 @@ def action_detailed_operations(self):
             }
         }
 
+    def _get_next_transfers(self):
+        next_transfers = set()
+        for move in self.move_ids:
+            for move_dest in move.move_dest_ids:
+                if move_dest.picking_id and move_dest.picking_id not in self.return_ids:
+                    next_transfers.add(move_dest.picking_id.id)
+        return next_transfers
```

Comment:
```suggestion
        next_pickings = self.move_ids.move_dest_ids.picking_id
        return next_pickings.filtered(lambda p: p not in self.return_ids)
```
The ORM is smart enough to set them inside a single recordset, so you don't have to iterate through them. :wink: 
Also, this way you return a recordset, which is easier to use afterwards.